src/libraries/System.Security.Cryptography.Xml/src/System/Security/Cryptography/Xml/CanonicalXmlDocument.cs

```diff
@@ -61,42 +61,42 @@ public void WriteHash(HashAlgorithm hash, DocPosition docPos, AncestralNamespace
             }
         }
 
-        public override XmlElement CreateElement(string prefix, string localName, string namespaceURI)
+        public override XmlElement CreateElement(string? prefix, string localName, string? namespaceURI)
         {
             return new CanonicalXmlElement(prefix, localName, namespaceURI, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlAttribute CreateAttribute(string prefix, string localName, string namespaceURI)
+        public override XmlAttribute CreateAttribute(string? prefix, string localName, string? namespaceURI)
         {
             return new CanonicalXmlAttribute(prefix, localName, namespaceURI, this, _defaultNodeSetInclusionState);
         }
 
-        protected override XmlAttribute CreateDefaultAttribute(string prefix, string localName, string namespaceURI)
+        protected override XmlAttribute CreateDefaultAttribute(string? prefix, string localName, string? namespaceURI)
         {
             return new CanonicalXmlAttribute(prefix, localName, namespaceURI, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlText CreateTextNode(string text)
+        public override XmlText CreateTextNode(string? text)
         {
             return new CanonicalXmlText(text, this, _defaultNodeSetInclusionState);
         }
 
-        public override XmlWhitespace CreateWhitespace(string prefix)
+        public override XmlWhitespace CreateWhitespace(string? prefix)
         {
-            return new CanonicalXmlWhitespace(prefix, this, _defaultNodeSetInclusionState);
+            return new CanonicalXmlWhitespace(prefix!, this, _defaultNodeSetInclusionState);
```

Comment:
this looks like potential NRE, might be good to open an issue (don't fix the product in this PR)

Reply:
It looks like `CanonicalXmlWhitespace` takes a nullable `prefix`. Have remove the bang from this line.

Reply:
`CanonicalXmlWhitespace` now takes a nullable `prefix`